Unable to build master following instructions in README.md

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

master branch

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

N/A

### Terraform Configuration Files

N/A

### Debug Output

N/A

### Panic Output

N/A

### Expected Behaviour

Following the instructions from README.md for building should work, allowing new contributors a stable base to build on. Currently instructions are basically `make tools && make build`

### Actual Behaviour

`make build` fails.

Initially it fails with lots of error messages of the form

```
$ make build
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
go generate ./azurerm/internal/services/...
panic: writing contents to "/tmptemp-1606686499.go": open /tmptemp-1606686499.go: permission denied

goroutine 1 [running]:
main.main()
	/home/jshare/go/src/github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tools/generator-resource-id/main.go:29 +0x294
exit status 2
azurerm/internal/services/analysisservices/resourceids.go:3: running "go": exit status 1

# panic repeated 30-40 times
```

Note the lack of a `/` between `/tmp` and `temp-xxxx.go`. I've temporarily hacked this locally with the following change

```diff
 func (f GolangCodeFormatter) randomFilePath() string {
        time := time.Now().Unix()
-       return fmt.Sprintf("%stemp-%d.go", os.TempDir(), time)
+       return fmt.Sprintf("%s/temp-%d.go", os.TempDir(), time)
 }
```

Although a better solution would probably use `filepath.Join` or `ioutil.TempFile`.

Running `make build` with this change I then stumble on the following error

```
$ make build
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
go generate ./azurerm/internal/services/...
go generate ./azurerm/internal/provider/
# github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/desktopvirtualization/parse
../services/desktopvirtualization/parse/application_group.go:73:6: undefined: strings
../services/desktopvirtualization/parse/host_pool.go:73:6: undefined: strings
azurerm/internal/provider/services.go:89: running "go": exit status 2
GNUmakefile:53: recipe for target 'generate' failed
make: *** [generate] Error 1
```

Looking at the git diff it seems that the generate step has _removed_ imports necesary from these files.

```
diff --git a/azurerm/internal/services/desktopvirtualization/parse/application_group.go b/azurerm/internal/services/desktopvirtualization/parse/application_group.go
index 1469e4692..789433d66 100644
--- a/azurerm/internal/services/desktopvirtualization/parse/application_group.go
+++ b/azurerm/internal/services/desktopvirtualization/parse/application_group.go
@@ -4,7 +4,6 @@ package parse
 
 import (
        "fmt"
-       "strings"
 
        "github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 )
diff --git a/azurerm/internal/services/desktopvirtualization/parse/host_pool.go b/azurerm/internal/services/desktopvirtualization/parse/host_pool.go
index 7fa736677..3fd339092 100644
--- a/azurerm/internal/services/desktopvirtualization/parse/host_pool.go
+++ b/azurerm/internal/services/desktopvirtualization/parse/host_pool.go
@@ -4,7 +4,6 @@ package parse
 
 import (
        "fmt"
-       "strings"
 
        "github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
 )

```

Fixing this manually obviously doesn't work as generate overwrites any fixes.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `git clone ...`
2. `make tools`
3. `make build`

### Important Factoids

Ubuntu 18.04.5 LTS

### References

From what I can tell from my own digging commit 84a7acfe53 is the source of both issues. (poke @tombuildsstuff )

### Additional Questions

Is there any reason why there isn't a github actions workflow confirming that the steps documented in the readme actually work?

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -17,6 +17,7 @@ tools:
 	GO111MODULE=off go get -u github.com/bflad/tfproviderlint/cmd/tfproviderlint
 	GO111MODULE=off go get -u github.com/bflad/tfproviderdocs
 	GO111MODULE=off go get -u github.com/katbyte/terrafmt
+	GO111MODULE=off go get -u golang.org/x/tools/cmd/goimports
 	GO111MODULE=off go get -u mvdan.cc/gofumpt
 	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH || $$GOPATH)/bin v1.32.0
 

